pkg/kubelet/cm/devicemanager/manager.go

```diff
@@ -840,6 +840,9 @@ func (m *ManagerImpl) allocateContainerResources(pod *v1.Pod, container *v1.Cont
 		if !m.isDevicePluginResource(resource) {
 			continue
 		}
+		if needed == 0 {
+			continue
+		}
```

Comment:
Should the pod be rejected by the API server if container requests <=0 devices?

Reply:
Apiserver is more focused on the scheme, and configuring `0` does not violate this rule